packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -0,0 +1,232 @@
+
+          /** @type {USDCProposalShapes['deposit']} */
+          // @ts-expect-error ensured by proposalShape
+          const proposal = lp.getProposal();
+          checkPoolBalance(poolSeat, shareWorth, USDC);
+          const post = depositCalc(shareWorth, proposal);
+
+          // COMMIT POINT
+
+          try {
+            const mint = shareMint.mintGains(post.payouts);
+            this.state.shareWorth = post.shareWorth;
+            zcf.atomicRearrange(
+              harden([
+                // zoe guarantees lp has proposal.give allocated
+                [lp, poolSeat, proposal.give],
+                // mintGains() above establishes that mint has post.payouts
+                [mint, lp, post.payouts],
+              ]),
+            );
+            lp.exit();
+            mint.exit();
+          } catch (cause) {
+            const reason = Error('🚨 cannot commit deposit', { cause });
+            console.error(reason.message, cause);
+            zcf.shutdownWithFailure(reason);
+          }
+          external.publishShareWorth();
+        },
+      },
+      withdrawHandler: {
+        /** @param {ZCFSeat} lp */
+        async handle(lp) {
+          const { shareWorth, shareMint, poolSeat } = this.state;
+          const { external } = this.facets;
+
+          /** @type {USDCProposalShapes['withdraw']} */
+          // @ts-expect-error ensured by proposalShape
+          const proposal = lp.getProposal();
+          const { zcfSeat: burn } = zcf.makeEmptySeatKit();
+          checkPoolBalance(poolSeat, shareWorth, USDC);
+          const post = withdrawCalc(shareWorth, proposal);
+
+          // COMMIT POINT
+
+          try {
+            this.state.shareWorth = post.shareWorth;
+            zcf.atomicRearrange(
+              harden([
+                // zoe guarantees lp has proposal.give allocated
+                [lp, burn, proposal.give],
+                // checkPoolBalance() + withdrawCalc() guarantee poolSeat has enough
+                [poolSeat, lp, post.payouts],
+              ]),
+            );
+            shareMint.burnLosses(proposal.give, burn);
+            lp.exit();
+            burn.exit();
+          } catch (cause) {
+            const reason = Error('🚨 cannot commit withdraw', { cause });
+            console.error(reason.message, cause);
+            zcf.shutdownWithFailure(reason);
+          }
+          external.publishShareWorth();
+        },
+      },
+      public: {
+        makeDepositInvitation() {
+          return zcf.makeInvitation(
+            this.facets.depositHandler,
+            'Deposit',
+            undefined,
+            this.state.proposalShapes.deposit,
+          );
+        },
+        makeWithdrawInvitation() {
+          return zcf.makeInvitation(
+            this.facets.withdrawHandler,
+            'Withdraw',
+            undefined,
+            this.state.proposalShapes.withdraw,
+          );
+        },
+        getPublicTopics() {
```

Comment:
I was surprised to see `getPublicTopics` on this Exo until I saw that this publicFacet is the publicFacet of the contract.

Consider making this public facet be its own `zone.exo` in the contract that calls out to a closely held `LiquidityPool` exo.

Reply:
Good idea, but after looking into it, I'd like to postpone this until we get more pieces of the contract together.

I started on it, but the LP exo isn't available until after remote calls, which would mean doing the same gymnastics for the public facet.